Editor confirmation sidebar: Update publishing state

jonathansadowski opened the pull request:
This PR sets the 'publishing' state upon confirmation in the sidebar. This closes the sidebar and displays a busy button that says "Publishing..." (or "Scheduling...") until the publish either succeeds or fails. The overlay is then closed.

![sidebar-publishing](https://cloud.githubusercontent.com/assets/363749/26805747/fe376f92-4a12-11e7-927b-daebfa0dccb3.gif)

This PR is part of a larger epic. The feature is behind a feature-flag and is intentionally incomplete. See p8F9tW-3F-p2.

To test:
* Checkout branch and run with the feature-flag enabled: `ENABLE_FEATURES=post-editor/delta-post-publish-flow make run`
* Load the editor and draft a post. Do not change the date.
* In Chrome you can throttle the connection speed from the network tab of developer tools. Select a slower speed such as GPRS.
![image](https://cloud.githubusercontent.com/assets/363749/26805805/63b06b62-4a13-11e7-88fd-29bb60ea65eb.png)
* If you don't use Chrome, throttling is possible in [Firefox](https://blog.nightly.mozilla.org/2016/11/07/simulate-slow-connections-with-the-network-throttling-tool/) or search for instructions to Throttle within your browser of choice.
* Click on "Publish". The sidebar should close and you should see "Publishing...".
* After the post has published, the overlay should disappear and you should see the current behavior (your post in the editor with a confirmation message)
* Repeat the process with a new post. Set the post to a future date. After clicking "Publish" in the confirmation dialog (that text to be updated in a future PR), you should see a busy button that reads "Scheduling...".

a8ci18n commented on client/post-editor/editor-confirmation-sidebar/index.jsx:
```diff
@@ -38,6 +39,21 @@ class EditorConfirmationSidebar extends React.Component {
+			case 'schedule':
+				return this.props.translate( 'Scheduling…' );
+			case 'publish':
+				return this.props.translate( 'Publishing…' );
```
Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'Publishing' )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/editor-confirmation/publishing-state)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/post-editor/editor-confirmation-sidebar/index.jsx:
```diff
@@ -38,6 +39,21 @@ class EditorConfirmationSidebar extends React.Component {
+			case 'update':
+				return this.props.translate( 'Updating…' );
+			case 'schedule':
+				return this.props.translate( 'Scheduling…' );
```
Hi! I've found a possible matching string that has already been translated **4** times:
`translate( 'Scheduling...' )` *ES Score: 25*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/post-editor/editor-confirmation-sidebar/index.jsx:
```diff
@@ -38,6 +39,21 @@ class EditorConfirmationSidebar extends React.Component {
+	getBusyButtonLabel( publishButtonStatus ) {
+		switch ( publishButtonStatus ) {
+			case 'update':
+				return this.props.translate( 'Updating…' );
```
Hi! I've found a possible matching string that has already been translated **39** times:
`translate( 'Updating...' )` *ES Score: 19*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/editor-confirmation/publishing-state)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/post-editor/editor-confirmation-sidebar/index.jsx:
```diff
@@ -38,6 +39,21 @@ class EditorConfirmationSidebar extends React.Component {
+			case 'schedule':
+				return this.props.translate( 'Scheduling...' );
+			case 'publish':
+				return this.props.translate( 'Publishing...' );
```
Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'Publishing' )` *ES Score: 18*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/editor-confirmation/publishing-state)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

michaeldcain commented on client/post-editor/editor-confirmation-sidebar/index.jsx:
```diff
@@ -38,6 +39,21 @@ class EditorConfirmationSidebar extends React.Component {
+	getBusyButtonLabel( publishButtonStatus ) {
+		switch ( publishButtonStatus ) {
+			case 'update':
+				return this.props.translate( 'Updating...' );
```
Looks like the linter wants an ellipsis (`opt+semicolon` on Mac) instead of three periods.

jonathansadowski replied:
It does, but @a8ci18n wants three dots because there's already an existing string "Updating...", I figured not created yet another string that wouldn't be translated was the better optimization.

jonathansadowski commented:
Thanks for the review @michaeldcain. The positioning got messed up when I did a rebase after originally submitting this PR. I've updated it and added a min-width to better ensure it covers the button below.
